packages/fuels-core/src/abi_decoder.rs

```diff
@@ -350,6 +327,27 @@ fn skip(slice: &[u8], num_bytes: usize) -> Result<&[u8]> {
     }
 }
 
+fn validate_not_nested_vector(param_type: &ParamType) -> Result<()> {
+    match param_type {
+        ParamType::Vector(..) => Err(error!(
+            InvalidData,
+            "Vectors containing vectors are not supported!"
+        )),
+        _ => Ok(()),
+    }
+}
```

Comment:
This will only detect immediately nested vectors, e.g. `Vec<Vec<u8>>` while not detecting if there is something separating them, for example in Sway: 
```rust
struct SomeStruct {
    a_field: u64,
    another_vector: Vec<u64>,
}
// later on in a contract
    fn bug() -> Vec<SomeStruct> {
        let mut inner_vec: Vec<u64> = Vec::new();
        inner_vec.push(1);
        inner_vec.push(2);
        inner_vec.push(3);

        let some_struct = SomeStruct {
            a_field: 4,
            another_vector: inner_vec,
        };

        let mut vec: Vec<SomeStruct> = Vec::new();
        vec.push(some_struct);

        vec
    }
```
when testing in our harness:
```rust
    {
        let expected: Vec<SomeStruct> = vec![SomeStruct {
            a_field: 4,
            another_vector: vec![1, 2, 3],
        }];
        let response = contract_methods.bug().call().await?.value;
        assert_eq!(response, expected);
    }
```
This would not fail but rather be populated by garbage data for the second vector.

Since vector decoding doesn't work for nested vectors, this validation needs to be fixed.

Reply:
Damn, nice catch, @segfault-magnet.

Reply:
'twas in tandem with @hal3e, which makes PR reviews super fun!

Reply:
Good catch, implementing a thorough recursive check as a method of `ParamType` and its tests.